src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,61 +1,70 @@
 // (C) 2020 GoodData Corporation
 import * as React from "react";
 import isFinite = require("lodash/isFinite");
-import { Execution } from "@gooddata/typings";
-import { stringToFloat } from "../../../helpers/utils";
-import { IntlWrapper } from "../../core/base/IntlWrapper";
 import { IntlTranslationsProvider, ITranslationsComponentProps } from "../../core/base/TranslationsProvider";
-import PushpinSizeLegend from "./legends/PushpinSizeLegend";
-import { getGeoData, getFormatFromExecutionResponse } from "../../../helpers/geoChart";
-import { IGeoConfig, IGeoData } from "../../../interfaces/GeoChart";
+import { IntlWrapper } from "../../core/base/IntlWrapper";
+
+import { IGeoConfig, IGeoData, IPushpinCategoryLegendItem } from "../../../interfaces/GeoChart";
 import { TOP } from "../../visualizations/chart/legend/PositionTypes";
-import { isTwoDimensionsData } from "../../../helpers/executionResultHelper";
+import PushpinSizeLegend, { IPushpinSizeLegendProps } from "./legends/PushpinSizeLegend";
+import PushpinCategoryLegend, {
+    IPushpinCategoryLegendProps,
+} from "../geoChart/legends/PushpinCategoryLegend";
 
 export interface IGeoChartLegendRendererProps {
     config: IGeoConfig;
-    execution: Execution.IExecutionResponses;
+    geoData: IGeoData;
+    sizeData?: number[];
+    segmentByData?: IPushpinCategoryLegendItem[];
+    sizeFormat?: string;
     locale: string;
     position?: string;
+    onItemClick?(item: IPushpinCategoryLegendItem): void;
 }
+
 export default function GeoChartLegendRenderer(props: IGeoChartLegendRendererProps): JSX.Element {
-    if (!props.execution) {
-        return null;
-    }
-    const {
-        execution: { executionResult, executionResponse },
-        config: { mdObject: { buckets = [] } = {} },
-        locale,
-        position = TOP,
-    } = props;
-    const geoData: IGeoData = getGeoData(buckets, executionResponse.dimensions);
-    const { data } = executionResult;
-    const { size } = geoData;
-    const classes = `geo-legend s-geo-legend position-${position}`;
+    const { position = TOP, segmentByData, sizeData } = props;
 
-    if (!size || !isTwoDimensionsData(data)) {
+    const classes = `geo-legend s-geo-legend position-${position}`;
+    if (!sizeData && !segmentByData) {
         return null;
     }
     return (
         <div className={classes}>
-            {size &&
-                renderPushpinSizeLegend(
-                    data[size.index].map(stringToFloat),
-                    getFormatFromExecutionResponse(size.index, executionResponse),
-                    locale,
-                )}
+            {segmentByData && renderPushpinCategoryLegend(props)}
+            {sizeData && renderPushpinSizeLegend(props)}
```

Comment:
default value of sizeData is empty array. So this condition isn't correct.

Reply:
fixed